### PR TITLE
add syntax sugar doc for static assignment

### PR DESCRIPTION
**Pull request by KLozes**
Related issue id = #574

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -58,4 +60,4 @@ The Taichi Programming Language
    :caption: Legacy
    :maxdepth: 3
 
-   installation
```
[the remark this answers is not in the document]
Comment (archibate) [reply docs/index.rst]: Vim doesn't like files without an EOF.. it force me to add one when save and I don't know how to turn it off.. Maybe we should let notepad/codeblocks/whatever users to self-discipline somehow..

[the remark this answers is not in the document]
Comment (archibate) [reply docs/index.rst]: Also make gcc happy to not generated warning sometime.

**Comment by KLozes** [comment]
@archibate good point. While I think objective data orient programing will be the most common use case, I should come up with something more general. perhaps:

ti.static() makes it possible to retain python style pointer assignment within Taichi kernels and functions. This allows local variables to alias global variables and functions, which can then be used throughout the kernel instead. 

show static assignment of global tensor and function...

This is also particularly useful for objective data-oriented programming, which inherits the python syntax of requiring self for class member access. 

show self simplification example ...

File: docs/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuanming-hu) [reply docs/index.rst]: Good point. I'll add this to the auto formatter.

**Pull request merged**